File: app/src/main/java/com/olehmesh/randomusers_task/adapters/ItemViewHolder.java
package com.olehmesh.randomusers_task.adapters;

import android.view.View;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.olehmesh.randomusers_task.R;
import com.olehmesh.randomusers_task.custom_views.CustomTextView;
import com.olehmesh.randomusers_task.models.Result;

public class ItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    OnItemClickListener onItemClickListener;

    Result list;

    @BindView(R.id.tvFirstName)
    CustomTextView tvName;
    @BindView(R.id.tvCity)
    CustomTextView tvCity;
    @BindView(R.id.ivAvatar)
    ImageView ivAvatar;


    public ItemViewHolder(@NonNull View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(this);
    }


    public interface OnItemClickListener {
        void onClick(View view, int position, boolean isLongClick);
    }

    public void setOnItemClickListener(OnItemClickListener itemClickListener) {
        this.onItemClickListener = itemClickListener;
    }


    @Override
    public void onClick(View view) {
        onItemClickListener.onClick(view, getAdapterPosition(), false);
    }
}

File: app/src/main/java/com/olehmesh/randomusers_task/adapters/ItemViewHolder.kt
package com.olehmesh.randomusers_task.adapters

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.Target
import com.olehmesh.randomusers_task.R
import com.olehmesh.randomusers_task.models.Result
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.main_list_item.*

class ItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), LayoutContainer {

    override val containerView: View?
        get() = itemView

    fun bind(item: Result) {

        tvFirstName.text = item.name!!.first
        tvCity.text = item.location!!.city

        Glide.with(itemView.context)
            .asBitmap()
            .load(item.picture!!.large)
            .apply(RequestOptions().override(Target.SIZE_ORIGINAL).encodeQuality(100))
            .placeholder(R.drawable.placeholder)
            .error(R.drawable.placeholder)
            .into(ivAvatar)

    }

}